mypy/main.py

```diff
@@ -31,6 +33,40 @@ def __init__(self) -> None:
         self.dirty_stubs = False
 
 
+class MyPyCommand(setuptools.Command):
+    """The :class:`MyPyCommand` class is used by setuptools to perform
+    type checks on registered modules.
+    """
+
+    description = "Validate PEP 0484 Types with mypy"
+    user_options = [
+        ('mypy-args=', None, 'Arguments to pass through to mypy')
+    ]  # type: List[tuple]
+
+    def initialize_options(self):
+        self.mypy_args = ''
+
+    def finalize_options(self):
+        pass
+
+    def run(self):
+        try:
+            mypy_options = []
+            for package in self.distribution.packages:
+                mypy_options.append('--package')
+                mypy_options.append(package)
+            mypy_options.extend(self.mypy_args.split())
+            sources, options = process_options(mypy_options)
+            if options.python_path and self.distribution.install_requires:
```

Comment:
I don't think I entirely follow this logic. Can you elaborate?

It looks like it says if we have the `python_path` option, then fetch all the dependencies that this distribution will need. I think I understand the point of fetching the dependencies here -- we're going to want to have them around for the type-checker to dig into. But what's the connection of that with the `python_path` option? If we have such an options, will the dependencies be fetched into a place where mypy will find them? If we don't, why does that mean we don't need to fetch them?

Very likely there are good answers to these questions for the code as written -- I'd just like to understand them. :-) And probably it'd be good to make a comment or something so that they can be understood by the next person reading this code.